File: WebApiRouter/Models/Endereco.cs
namespace WebApiRouter.Models;

public partial class Endereco
{
    public int Id { get; set; }

    public string Logradouro { get; set; } = null!;

    public string Numero { get; set; } = null!;

    public string Bairro { get; set; } = null!;

    public string Cidade { get; set; } = null!;

    public string Estado { get; set; } = null!;

    public sbyte Ativo { get; set; }

    public virtual ICollection<Pessoajuridica> Pessoajuridicas { get; set; } = new List<Pessoajuridica>();
}

File: routerapp/lib/models/endereco.dart
class Endereco {
  final String logradouro;
  final String numero;
  final String bairro;
  final String cidade;
  final String estado;

  Endereco({
    required this.logradouro,
    required this.numero,
    required this.bairro,
    required this.cidade,
    required this.estado,
  });

  factory Endereco.fromJson(Map<String, dynamic> json) {
    return Endereco(
      logradouro: json['logradouro'],
      numero: json['numero'],
      bairro: json['bairro'],
      cidade: json['cidade'],
      estado: json['estado'],
    );
  }
}
